bugrap/src/main/java/org/vaadin/bugrap/ui/ReportsView.java

```diff
@@ -47,24 +67,81 @@ public void enter(ViewChangeEvent event) {
+		
+		menuStatus.getItems().get(MENUBARIDX_STATUS_OPEN).setCommand(evt -> onStatusFilterSelected(true, true, Status.OPEN));
+		menuStatus.getItems().get(MENUBARIDX_STATUS_ALL).setCommand(evt -> onStatusFilterSelected(true, true, Status.values()));
+		
+		MenuItem customStatusMenuItem = menuStatus.getItems().get(MENUBARIDX_STATUS_CUSTOM);
+		customStatusMenuItem.getChildren().get(MENUBARIDX_CUSTOMSTATUS_OPEN).setCommand(evt -> onStatusFilterSelected(false, evt.isChecked(), Status.OPEN));
+		customStatusMenuItem.getChildren().get(MENUBARIDX_CUSTOMSTATUS_CANTFIX).setCommand(evt -> onStatusFilterSelected(false, evt.isChecked(), Status.CANT_FIX));
+		customStatusMenuItem.getChildren().get(MENUBARIDX_CUSTOMSTATUS_DUPLICATE).setCommand(evt -> onStatusFilterSelected(false, evt.isChecked(), Status.DUPLICATE));
+		customStatusMenuItem.getChildren().get(MENUBARIDX_CUSTOMSTATUS_FIXED).setCommand(evt -> onStatusFilterSelected(false, evt.isChecked(), Status.FIXED));
+		customStatusMenuItem.getChildren().get(MENUBARIDX_CUSTOMSTATUS_INVALID).setCommand(evt -> onStatusFilterSelected(false, evt.isChecked(), Status.INVALID));
+		customStatusMenuItem.getChildren().get(MENUBARIDX_CUSTOMSTATUS_NEEDSINFO).setCommand(evt -> onStatusFilterSelected(false, evt.isChecked(), Status.NEED_MORE_INFO));
+		customStatusMenuItem.getChildren().get(MENUBARIDX_CUSTOMSTATUS_WONTFIX).setCommand(evt -> onStatusFilterSelected(false, evt.isChecked(), Status.WONT_FIX));
+		customStatusMenuItem.getChildren().get(MENUBARIDX_CUSTOMSTATUS_WORKS4ME).setCommand(evt -> onStatusFilterSelected(false, evt.isChecked(), Status.WORKS_FOR_ME));
+	}
+
+	private void onAssigneeFilterSelected(int mode) {
+		model.setAssigneeFilterMode(mode);
+		drawAssigneeMenubarFromModel();
+		refreshGridContent();
+	}
+
+	private void onStatusFilterSelected(boolean isSetMode, boolean isChecked, Status...status) {
+		if (isSetMode)
+			model.setStatusFilters(status);
+		else
+			model.changeStatusFilters(isChecked, status);
+		drawStatusMenubarFromModel();
+		refreshGridContent();
+	}
+	
+	private void drawAssigneeMenubarFromModel() {
+		menuAssignees.getItems().get(MENUBARIDX_ASSIGNEE_ME).setChecked(model.getAssigneeFilterMode() == ReportsModel.ASSIGNEE_ME);
+		menuAssignees.getItems().get(MENUBARIDX_ASSIGNEE_ALL).setChecked(model.getAssigneeFilterMode() == ReportsModel.ASSIGNEE_ALL);
+	}
+
+	private void drawStatusMenubarFromModel() {
+		menuStatus.getItems().get(MENUBARIDX_STATUS_OPEN).setChecked(model.getStatusFilter().size() == 1 && model.getStatusFilter().contains(Status.OPEN));
+		menuStatus.getItems().get(MENUBARIDX_STATUS_ALL).setChecked(model.getStatusFilter().containsAll(Arrays.asList(Status.values())));
+		
+		MenuItem customStatusMenuItem = menuStatus.getItems().get(MENUBARIDX_STATUS_CUSTOM);
+		customStatusMenuItem.getChildren().get(MENUBARIDX_CUSTOMSTATUS_OPEN).setChecked(model.getStatusFilter().contains(Status.OPEN));
```

Comment:
getChildren() could be called only once.

Consider other places where this comment could be applied.